vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinServletContextInitializer.java

```diff
@@ -294,9 +296,16 @@ public void failFastContextInitialized(ServletContextEvent event) {
             Stream<Class<? extends Component>> hasErrorComponents = findBySuperType(
                     getErrorParameterPackages(), HasErrorParameter.class)
                             .filter(Component.class::isAssignableFrom)
+                            // Replace Flow default with custom version for Spring
+                            .filter(clazz -> clazz != RouteNotFoundError.class)
                             .map(clazz -> (Class<? extends Component>) clazz);
-            registry.setErrorNavigationTargets(
-                    hasErrorComponents.collect(Collectors.toSet()));
+            final Set<Class<? extends Component>> errorComponents =
+                    hasErrorComponents.collect(Collectors.toSet());
+
+            if (errorComponents.stream().noneMatch(RouteNotFoundError.class::isAssignableFrom)) {
+                errorComponents.add(SpringRouteNotFoundError.class);
+            }
+            registry.setErrorNavigationTargets(errorComponents);
```

Comment:
I would also create a test for this regression, to be safe.

Reply:
I agree, either unit test or manual testing needed. Can this be unit tested ?

Reply:
Unit test added.